compiler/noirc_errors/src/position.rs

```diff
@@ -8,6 +8,61 @@ use std::{
 
 pub type Position = u32;
 
+#[derive(Eq, Debug, Clone)]
+pub struct Located<T> {
+    pub contents: T,
+    location: Location,
+}
+
+/// This is important for tests. Two Located objects are equal if their content is equal
+/// They may not have the same location. Use into_location to test for Location being equal specifically
+impl<T: PartialEq> PartialEq<Located<T>> for Located<T> {
+    fn eq(&self, other: &Located<T>) -> bool {
+        self.contents == other.contents
+    }
+}
+
+impl<T: PartialOrd> PartialOrd<Located<T>> for Located<T> {
+    fn partial_cmp(&self, other: &Located<T>) -> Option<std::cmp::Ordering> {
+        self.contents.partial_cmp(&other.contents)
+    }
+}
+
+impl<T: Ord> Ord for Located<T> {
+    fn cmp(&self, other: &Located<T>) -> std::cmp::Ordering {
+        self.contents.cmp(&other.contents)
+    }
+}
+
+impl<T: Default> Default for Located<T> {
+    fn default() -> Self {
+        Self { contents: Default::default(), location: Location::dummy() }
+    }
+}
+
+/// Hash-based data structures (HashMap, HashSet) rely on the inverse of Hash
+/// being injective, i.e. x.eq(y) => hash(x, H) == hash(y, H), we hence align
+/// this with the above
+impl<T: Hash> Hash for Located<T> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.contents.hash(state);
+    }
+}
+
+impl<T> Located<T> {
+    pub const fn from(location: Location, contents: T) -> Located<T> {
+        Located { location, contents }
+    }
```

Comment:
Do we require this to be const currently? Or is there an advantage in it being const if we're not using it in a const context?

Reply:
Sure, lets remove it. It probably doesn't do anything but when I see it it makes me think "wait, do we use this in a const context anywhere?"